<launch>

    <!-- Param --> 

    <param name="acca_folder" type="str" value="/home/acca/catkin_ws/src/ACCA" />
    <param name="odometry_topic" type="str" value="/erp42_fusion_odometry" />
    <!-- Stanley Control Params -->
    <!-- <param name="initial_idx" type="int" value="0" />
    <param name="c_gain" type="double" value="1.0" />
    <param name="hdr_ratio" type="double" value="1.0" /> -->
    <!-- Pure_pursuit Params-->
    <!-- <param name=""> -->
    <!-- SPEED Settings -->
    <param name="desired_speed" type="double" value="7.0" />
    <!-- Parking Params -->
    <param name="max_steer" type="double" value="50.0" />
    <!-- RVIZ -->
    
    
    <!-- LiDAR -->
    
    <!-- <include file="$(find velodyne_pointcloud)/launch/VLP16_points.launch"/> -->
    <!-- <include file="$(find adaptive_clustering)/launch/adaptive_clustering.launch"/> -->
    <!-- <node pkg="ma_rrt_path_plan" type="3dcone.py" name="cone_to_track" /> -->


    <!-- lms1xx -->
    <node pkg="lms1xx" type="LMS1xx_node" name="lms1xx" output="screen">
        <param name="sensor_param" value="param_value" />
    </node>
    <node pkg="laser_filters" type="scan_to_scan_filter_chain" name="laser_filter">
        <rosparam command="load" file="$(find laser_filters)/my_laser_config.yaml" />
    </node>
    <node pkg="tf" type="static_transform_publisher" name="laser_pub" args="0.52 -0.2 0.0 0.0 0.0 0.0 base_link laser 10" />
    
    <!-- IMU -->

    <include file="$(find xsens_mti_driver)/launch/xsens_mti_node.launch" >
        <arg name="port" value="/dev/ttyUSB1" />
    </include>

    <!-- Cone Tracker -->

    <node pkg="cone_tracker" type="dbscan_right.py" name="dbscan"/>      
    <node pkg="cone_tracker" type="cone_to_track.py" name="cone_to_track"/>
    <node pkg="cone_tracker" type="waypointsmaker_pp_right.py" name="waypointsmaker" output="screen"/>
    <!-- <node pkg="cone_tracker" type="publish_cone_path.py" name="publish_cone_path" /> -->

    <!-- <node pkg="tf" type="static_transform_publisher" name="odom_to_laser" args="0.5 0 0 0 0 0 odom base_link 10"/> -->

    <!-- <node pkg="ma_rrt_path_plan" type="main.py" name="ma_rrt_path_plan_node" output="screen">
        <param name="desiredWaypointsFrequency" value="10" type="int" />
        <param name="odom_topic" value="/erp42_fusion_odometry"/> -->
        <!-- <param name="odom_topic" value="odometry/filtered"/> -->
        <!-- <param name="world_frame" value="odom"/>
        <param name="publishWaypoints" type="bool" value="true" />
    </node> -->


    <!-- Odometry -->


    <param name="erp_port" type="str" value="/dev/ttyUSB0" />
    <node name="erp42_encoder" pkg="odometry" type="erp42_encoder.py" />
    <node name="erp42_odometry" pkg="odometry" type="erp42_fusion_odometry.py" />
    <!-- <include file="$(find robot_localization)/launch/ekf.launch" /> -->
    <!-- <include file="$(find lio_sam)/launch/run.launch"/> -->
    
    <include file="$(find erp42_vehicle_rviz)/launch/erp42_vehicle_rviz.launch">
        <arg name="rvizconfig" value="/home/acca/dddd.rviz" />
    </include>
</launch>
